### DISCUSSION TITLE
Compile flatpak old Freerdp version

### DISCUSSION THREAD

**Discussion by pes00** [FreeRDP/FreeRDP#10698] [Q&A]
I would like to compile Freerdp version 2.11.2 for Flatpak.
I have installed flatpak and flatpak-builder
I have read the wiki and the related commands 
https://github.com/FreeRDP/FreeRDP/wiki/Compilation#compilation-1
but since I have no experience I can't. I have also searched for examples on the internet but with no luck.
Can someone help me with the exact command to run?

**Comment by akallabeth** [comment]
@pes00 just run the commands for flatpak documented there from the checkout you did?

**Reply by pes00** [reply]
I run command:
`flatpak-builder --repo=repo /home/pi/FreeRDP/ packaging/flatpak/com.freerdp.FreeRDP.json`
and I got the following error
`Can't load 'packaging/flatpak/com.freerdp.FreeRDP.json': Apertura del file Â«/home/pi/packaging/flatpak/com.freerdp.FreeRDP.jsonÂ» non riuscita: File o directory non esistente`

**Reply by akallabeth** [reply]
well, I´d read the line again then ;)
1. you must run it from the freerdp source directory **NOT** your user home directory
2. you should provide a build directory to the command (that should **NOT** be your source directory)

**Reply by akallabeth** [reply]
quoting from the referenced wiki:

> Run flatpak-builder --repo=repo \<build dir\> packaging/flatpak/com.freerdp.FreeRDP.json from the checkout root

**Reply by pes00** [reply]
These are the steps performed:
-downloaded freerdp version 2.11.2
`sudo git clone https://github.com/FreeRDP/FreeRDP.git --branch 2.11.2`

-then I compile
`flatpak-builder --repo=repo /home/pi/FreeRDPFlatpak/ packaging/flatpak/com.freerdp.FreeRDP.json`
and then 
`flatpak build-bundle repo com.freerdp.FreeRDP.flatpak com.freerdp.FreeRDP --runtime repo=https://flathub.org/repo/flathub.flatpakrepo`

-then I install
`flatpak install com.freerdp.FreeRDP.flatpak`

but when I run
`flatpak run com.freerdp.FreeRDP /u:user /p:password /v:111.111.111.111 /workarea -decorations /cert:ignore /network:auto`

I get the following error
`XDG_SESSION_TYPE undefined`

did I do something wrong?

**Reply by akallabeth** [reply]
> These are the steps performed: -downloaded freerdp version 2.11.2 `sudo git clone https://github.com/FreeRDP/FreeRDP.git --branch 2.11.2`
> 

first error, never use `sudo` for something like a git checkout/clone

> 
> I get the following error `XDG_SESSION_TYPE undefined`
> 

well, where are you running your session from?
try running it with `XDG_SESSION_TYPE=x11 flatpak run com.freerdp.FreeRDP ...`

`XDG_SESSION_TYPE` is defined for all `X11` and `wayland ` environments so this sounds like a bug in your environment ;)

**Reply by pes00** [reply]
I have no hope. Now a new error appears. 
`X11 /app/bin/xfreerdp
[13:35:25:011] [4:5] [ERROR][com.freerdp.codec] - Failed to initialize libav parser
[13:35:25:011] [4:5] [ERROR][com.freerdp.core.codecs] - Failed to create h264 codec context
[13:35:25:014] [4:4] [ERROR][com.freerdp.core] - freerdp_abort_connect:freerdp_set_last_error_ex ERRCONNECT_CONNECT_CANCELLED [0x0002000B]
`
Too complicated for me. Patience. I'll wait for when you can analyze the "workarea" problem
